smac/facade/smac_facade.py

```diff
@@ -32,7 +33,9 @@ class SMAC(object):
 
     def __init__(self,
                  scenario: Scenario,
-                 tae_runner: ExecuteTARun=None,
+                 # TODO: once we drop python3.4 add type hint
```

Comment:
Ok, let's be more concrete: I vote for Python 3.5 support only. The only larger project that depends on SMAC3 is auto-sklearn right now, right?  And you are going to drop Python 2.7 soon -- so you could go for Python 3.5 straight away.

Furthermore, I would expect that either we will drop Python 3.4 support soon or we keep it for a long time. 
And I don't see a reason for a Python 3.4 support right now. We decided to go for Python 3 in SMAC3 because it is the default in Ubuntu 16.04, but in fact it is Python 3.5 and not 3.4 in there.

Last but not least, could you please also fix the readme and remove the sentence: 

> We expect that it should be also compatible with Python 2.7.